packages/office-ui-fabric-react/src/components/Slider/Slider.base.tsx

```diff
@@ -46,27 +38,30 @@ export class SliderBase extends BaseComponent<ISliderProps, ISliderState> implem
 
     this.state = {
       value: value,
-      renderedValue: value
+      renderedValue: undefined
     };
   }
 
-  /**
-   * Invoked when a component is receiving new props. This method is not called for the initial render.
-   */
-  public componentWillReceiveProps(newProps: ISliderProps): void {
-    if (newProps.value !== undefined) {
-      const value = Math.max(newProps.min as number, Math.min(newProps.max as number, newProps.value));
-
-      this.setState({
-        value: value,
-        renderedValue: value
-      });
-    }
-  }
-
   public render(): React.ReactElement<{}> {
-    const { ariaLabel, className, disabled, label, max, min, showValue, buttonProps, vertical, valueFormat, styles, theme } = this.props;
-    const { value, renderedValue } = this.state;
+    const {
+      ariaLabel,
+      className,
+      disabled,
+      label,
+      max,
+      min,
+      showValue,
+      buttonProps,
+      vertical,
+      valueFormat,
+      styles,
+      theme,
+      value: propsValue
+    } = this.props;
+    const { value: stateValue, renderedValue: renderedValueState } = this.state;
+    const value = propsValue !== undefined ? propsValue : stateValue;
+    const renderedValue = renderedValueState !== undefined ? renderedValueState : value;
```

Comment:
Simpler(?):
```ts
const { value = this.state.value } = this.props;
const { renderedValue = value } = this.state;
```

Reply:
And you should probably add a comment about the props value overriding the state value being intentional.

Reply:
Hmm actually I'd recommend the change above for renderedValue, but for value, you should update the `value` getter on the class and use it everywhere (instead of directly accessing state or props value):
```
public get value(): number | undefined {
  const { value = this.state.value } = this.props;
  return value;
}
```

Reply:
Very good idea! This also helped me catch and enforce the value with respect to min and max.